controllers/front/Payment.php

```diff
@@ -412,4 +421,94 @@ public function buildRecurringData($request = array(), $payload)
         return $request;
     }
 
+    private function buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request = array())
+    {
+        $request['paymentMethod']['type'] = $paymentType;
+        if (!empty($paymentIssuer)) {
+            $request['paymentMethod']['issuer'] = $paymentIssuer;
+        }
+
+        return $request;
+    }
+
+    private function buildReturnURL($request)
+    {
+        $request['returnUrl'] = $this->context->link->getModuleLink(
+            $this->module->name,
+            'Result',
+            array(),
+            $this->ssl
+        );
+
+        return $request;
+    }
+
+    private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIssuer)
+    {
+        $request = array();
+        $request = $this->buildBrowserData($request);
+        $request = $this->buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request);
+        $request = $this->buildPaymentData($request);
+        $request = $this->buildMerchantAccountData($request);
+        $request = $this->buildRecurringData($request, $_REQUEST);
```

Comment:
I think we don't need this one

Reply:
We do. The `$request` array is just created there.

Reply:
buildRecurringData fills the request only if 'storeCc' is true, which is the checkbox we use for saving the credit cards, we don't need this for local payment methods. I am referring only to the buildRecurringData, the others are needed oc.

Reply:
```suggestion
```